test/extended/builds/valuefrom.go

```diff
@@ -0,0 +1,210 @@
+package builds
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+
+)
+
+var _ = g.Describe("[builds][valueFrom] process valueFrom in build strategy environment variables", func() {
+	var (
+		valueFromBaseDir               = exutil.FixturePath("testdata", "valuefrom")
+		dockerImageStreamFixture       = filepath.Join(valueFromBaseDir, "docker-is.json")
+		stiImageStreamFixture          = filepath.Join(valueFromBaseDir, "sti-is.json")
+		secretFixture                  = filepath.Join(valueFromBaseDir, "test-secret.yaml")
+		configmapFixture               = filepath.Join(valueFromBaseDir, "test-configmap.yaml")
+		successfulSTIBuildValueFrom    = filepath.Join(valueFromBaseDir, "successful-sti-build-value-from-config.yaml")
+		successfulDockerBuildValueFrom = filepath.Join(valueFromBaseDir, "successful-docker-build-value-from-config.yaml")
+		failedSTIBuildValueFrom        = filepath.Join(valueFromBaseDir, "failed-sti-build-value-from-config.yaml")
+		failedDockerBuildValueFrom     = filepath.Join(valueFromBaseDir, "failed-docker-build-value-from-config.yaml")
+		failedSTIBuildEnvReference     = filepath.Join(valueFromBaseDir, "failed-sti-build-env-reference-config.yaml")
+		failedDockerBuildEnvReference  = filepath.Join(valueFromBaseDir, "failed-docker-build-env-reference-config.yaml")
+		oc                             = exutil.NewCLI("build-valuefrom", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("waiting for openshift namespace imagestreams")
+		err = exutil.WaitForOpenShiftNamespaceImageStreams(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should successfully process valueFrom in s2i build strategy environment variables", func() {
+
+		g.By("creating test image stream")
+		err := oc.Run("create").Args("-f", stiImageStreamFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test secret")
```

Comment:
push this and the "creating test configmap" into g.JustBeforeEach() above to save duplication?